test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Formatter/JsonLight/Metadata/ComplexTypeTests.cs

```diff
@@ -85,5 +86,38 @@ protected static IEdmModel GetEdmModel(WebRouteConfiguration configuration)
                 }
             }
         }
+
+        [Theory]
+        [InlineData("application/json;odata.metadata=full", "/Complex/EntityWithComplexProperties?$select=StringListProperty")]
+        [InlineData("application/json;odata.metadata=full", "/Complex/EntityWithComplexProperties?$select=ComplexProperty")]
+        [InlineData("application/json;odata.metadata=full", "/Complex/EntityWithComplexProperties?$count=true")]
+        [InlineData("application/json;odata.metadata=full", "/Complex/EntityWithComplexProperties?$count=true&$select=ComplexProperty")]
```

Comment:
The title of your PR says AddODataQueryFilter. Is the $filter implementation not included for this test case?

Reply:
The AddODataQueryFilter is for enable the service configuration for post-execution query processing. It is not really for $filter.